tests/cli/test_model_hook_order.py

```diff
@@ -0,0 +1,107 @@
+from ert_shared.models import (
+    EnsembleSmoother,
+    MultipleDataAssimilation,
+    IteratedEnsembleSmoother,
+)
+import inspect
+
+from unittest.mock import MagicMock
+
+EXPECTED_CALL_ORDER = [
+    "PRE_SIMULATION",
+    "POST_SIMULATION",
+    "PRE_FIRST_UPDATE",
+    "PRE_UPDATE",
+    "POST_UPDATE",
+    "PRE_SIMULATION",
+    "POST_SIMULATION",
+]
+
+
+def test_hook_call_order_ensemble_smoother(monkeypatch):
+    test_class = EnsembleSmoother
+    mock_parent = MagicMock()
+    mock_sim_runner = MagicMock()
+    mock_parent.runWorkflows = mock_sim_runner
+    test_module = inspect.getmodule(test_class)
+    monkeypatch.setattr(test_module, "EnkfSimulationRunner", mock_parent)
+    monkeypatch.setattr(test_module, "ERT", MagicMock())
+    test_class.runSimulations(MagicMock(), MagicMock())
+    assert mock_sim_runner.mock_calls
+    for expected_call, actual_call in zip(
```

Comment:
Could this be a good application of `assert_has_calls`? See https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls

Reply:
The main reason I didn´t is because while order needs to be preserved for `assert_has_calls`, the list does not have to equal, and I intent to assert that the first element is the specified one. For example is this valid:

```
from mock import Mock, call
mock = Mock(return_value=None)
mock(1)
mock(2)
mock(3)
calls = [call(2), call(3)]
mock.assert_has_calls(calls)
```

Can change to asserting that the `calls` are there though, but not sure that adds that much value? Will do it if you would like though.

Reply:
Changed to asserting that `mock_calls` are the expected calls. That makes it easier to debug as well, thanks for the suggestion 👍 

Would you prefer that I check against `assert_has_calls` and also length instead?

Reply:
Cool 😎